olimex/teres_i/default.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ...}:
+
+{
+  hardware.deviceTree = {
+    name = lib.mkDefault "allwinner/sun50i-a64-teres-i.dts";
```

Comment:
This is not needed, as it is used to pick a different `dtb` filename from the default in the extlinux-compatible boot flow.

Setting it *could* (but probably won't) break things as it would not naturally change if the platform firmware + kernel pairing of DTB filenames change for the board.

Reply:
See rationale in https://github.com/NixOS/nixos-hardware/pull/750#discussion_r1349559992

tldr: it's needed as the bootloader likes to sometimes prefer generic a64 sunxi dtb over the one specific to teres

Reply:
Only mainline U-Boot is aimed to be supported, unless specified otherwise.

Mainline U-Boot will use the appropriate dtb filename here.

If it does not, you are not using mainline U-Boot, or have it misconfigured. (There is the unlikely possibility of a bug upstream, but if it did, every board at least using the same SoC would be broken, if not all boards using a dtb).

Reply:
i do more tests on this and get back to you